benchmarks/Bio/BCR.jmd

```diff
@@ -74,78 +81,200 @@ show(to)
+
+#### Plot Work-Precision Diagram (using GMRES linear solver).
+
+```julia
+wp = WorkPrecisionSet(oprob,abstols,reltols,setups;error_estimate=:l2,
+                    saveat=tf/10000.,appxsol=test_sol,maxiters=Int(1e12),dtmin=1e-18,numruns=2)
+
+names = ["TRBDF2 (GMRES)" "QNDF (GMRES)" "FBDF (GMRES)" "KenCarp4 (GMRES)"]
+xlimit,ylimit = plot_settings(wp)
+plot(wp;label=names,xlimit=xlimit,ylimit=ylimit)
 ```
 
-## Sparse Jacobian
-Finally we test using the generated sparse analytic Jacobian function.
+#### Declare solvers (using GMRES linear solver, with pre-conditioner).
+
 ```julia
 setups = [
-          #Dict(:alg=>Rosenbrock23(autodiff=false)),
-          Dict(:alg=>TRBDF2(autodiff=false)),
-          Dict(:alg=>QNDF(autodiff=false)),
-          Dict(:alg=>FBDF(autodiff=false)),
-          #Dict(:alg=>CVODE_BDF(linear_solver=:KLU)), # Fails!
-          #Dict(:alg=>rodas()),
-          #Dict(:alg=>radau()),
-          #Dict(:alg=>Rodas4(autodiff=false)),
-          #Dict(:alg=>Rodas5(autodiff=false)),
-          Dict(:alg=>KenCarp4(autodiff=false)),
-          Dict(:alg=>KenCarp47(autodiff=false)),
-          #Dict(:alg=>RadauIIA5(autodiff=false)),
-          #Dict(:alg=>lsoda()),
-          ];
+        Dict(:alg=>TRBDF2(linsolve=KrylovJL_GMRES(),autodiff=false,precs=incompletelu_fceri,concrete_jac=true)),
+        Dict(:alg=>QNDF(linsolve=KrylovJL_GMRES(),autodiff=false,precs=incompletelu_fceri,concrete_jac=true)),
+        Dict(:alg=>FBDF(linsolve=KrylovJL_GMRES(),autodiff=false,precs=incompletelu_fceri,concrete_jac=true)),
+        Dict(:alg=>KenCarp4(linsolve=KrylovJL_GMRES(),autodiff=false,precs=incompletelu_fceri,concrete_jac=true))
+        ];
+```
+
+#### Plot Work-Precision Diagram (using GMRES linear solver, with pre-conditioner).
+
+```julia
+wp = WorkPrecisionSet(oprob_sparse,abstols,reltols,setups;error_estimate=:l2,
+                    saveat=tf/10000.,appxsol=test_sol,maxiters=Int(1e12),dtmin=1e-18,numruns=2)
+
+names = ["TRBDF2 (GMRES, iLU)" "QNDF (GMRES, iLU)" "FBDF (GMRES, iLU)" "KenCarp4 (GMRES, iLU)"]
+xlimit,ylimit = plot_settings(wp)
+plot(wp;label=names,xlimit=xlimit,ylimit=ylimit)
+```
+
+#### Declare solvers (using sparse jacobian)
+
+We designate the solvers we wish to use.
+```julia
+setups = [
+        Dict(:alg=>TRBDF2(linsolve=KLUFactorization(),autodiff=false)),
+        Dict(:alg=>QNDF(linsolve=KLUFactorization(),autodiff=false)),
+        Dict(:alg=>FBDF(linsolve=KLUFactorization(),autodiff=false)),
+        Dict(:alg=>KenCarp4(linsolve=KLUFactorization(),autodiff=false))
+        ];
+```
+
+
+#### Plot Work-Precision Diagram (using sparse jacobian)
+
+Finally, we generate a work-precision diagram for the selection of solvers.
+```julia
 wp = WorkPrecisionSet(sparsejacprob,abstols,reltols,setups;error_estimate=:l2,
-                      saveat=tf/10000.,appxsol=test_sol,maxiters=Int(1e5),numruns=1)
-plot(wp)
+                    saveat=tf/10000.,appxsol=test_sol,maxiters=Int(1e12),dtmin=1e-18,numruns=2)

```

Comment:
why was this max iters increased so high? That shows that one of the solvers is probably actually failing and should be removed from the set.